presto-main-base/src/test/java/com/facebook/presto/security/TestAccessControlManager.java

```diff
@@ -258,6 +261,67 @@ public void testDenySystemAccessControl()
+        TransactionManager transactionManager = createTestTransactionManager(catalogManager);
+        AccessControlManager accessControlManager = new AccessControlManager(transactionManager);
+
+        accessControlManager.addSystemAccessControlFactory(new SystemAccessControlFactory() {
+            @Override
+            public String getName()
+            {
+                return "test";
+            }
+
+            @Override
+            public SystemAccessControl create(Map<String, String> config)
+            {
+                return new SystemAccessControl() {
+                    @Override
+                    public Map<ColumnMetadata, ViewExpression> getColumnMasks(Identity identity, AccessControlContext context, CatalogSchemaTableName tableName, List<ColumnMetadata> columns)
+                    {
+                        ImmutableMap.Builder<ColumnMetadata, ViewExpression> columnMaskBuilder = ImmutableMap.builder();
+                        for (ColumnMetadata column : columns) {
+                            columnMaskBuilder.put(column, new ViewExpression("user", Optional.empty(), Optional.empty(), "system mask"));
+                        }
+                        return columnMaskBuilder.buildOrThrow();
+                    }
+
+                    @Override
+                    public void checkCanSetUser(Identity identity, AccessControlContext context, Optional<Principal> principal, String userName)
+                    {
+                    }
+
+                    @Override
+                    public void checkQueryIntegrity(Identity identity, AccessControlContext context, String query)
+                    {
+                    }
+
+                    @Override
+                    public void checkCanSetSystemSessionProperty(Identity identity, AccessControlContext context, String propertyName)
+                    {
+                    }
+                };
+            }
+        });
+        accessControlManager.setSystemAccessControl("test", ImmutableMap.of());
+
+        ConnectorId connectorId = registerBogusConnector(catalogManager, transactionManager, accessControlManager, "catalog");
+        accessControlManager.addCatalogAccessControl(connectorId, new ConnectorAccessControl() {
+            @Override
+            public Map<ColumnMetadata, ViewExpression> getColumnMasks(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, SchemaTableName tableName, List<ColumnMetadata> columns)
+            {
+                ImmutableMap.Builder<ColumnMetadata, ViewExpression> columnMaskBuilder = ImmutableMap.builder();
+                for (ColumnMetadata column : columns) {
+                    columnMaskBuilder.put(column, new ViewExpression("user", Optional.empty(), Optional.empty(), "connector mask"));
+                }
+                return columnMaskBuilder.buildOrThrow();
+            }
+        });
+    }
```

Comment:
Maybe I am , but I don't actually see any assertions in this test. What are we testing here?

Reply:
Yeah, only the assertions were removed in this cherry-pick https://github.com/trinodb/trino/commit/bdd1cb5dbc66e938de4802145ee75f5cfb1e4bb1#diff-909e4c713f94af653c076fb89d41721b93d272cc80664238535e3f572c976a79

Reply:
Since multiple masks are not allowed on a column, this test now throws an exception. There aren't any other tests that check this behavior, so I'll convert it to verify that it throws.

Reply:
done